ports/tcpport.js

```diff
@@ -118,6 +118,12 @@ TcpPort.prototype.write = function(data) {
     // get next transaction id
     var transactionsId = (this._transactionId + 1) % MAX_TRANSACTIONS;
 
+    // remember current unit and command
+    this._id = data[0];
+    this._cmd = data[1];
+
+    // TODO: is it not interesting to have the expected length here as in all other ports?
```

Comment:
A. In the case of modbus tcp, we are not guaranteed a `request -> response` order, like other ports.

For example we can have a `request_12 `and then get `response_1` and then `response_4` and then `request_13` ... 
We can not trust the values in `this._id` , `this._cmd` and/or `this._length` to correspond to the next response.

B. We save the `transactionId` , so we can store this values for this transaction, and retrieve the appropriate values once we read the `transactionId` of the response packet.

C. I think it's not safe to store `this._id` and `this._cmd` here, because the next response may not be the response for this request. We do not test for this sequence of events ... so the tests will not catch that, but it's wrong.